src/platforms/rp2040/src/lib/networkdriver.c

```diff
@@ -235,8 +260,53 @@ static term start_sta(term sta_config, GlobalContext *global)
             return BADARG_ATOM;
         }
     }
+    if (term_is_invalid_term(hostname_term)) {
+        size_t buf_size = DEFAULT_HOSTNAME_SIZE;
+        driver_data->hostname = malloc(buf_size);
+        if (IS_NULL_PTR(driver_data->hostname)) {
+            free(ssid);
+            free(psk);
+            return OUT_OF_MEMORY_ATOM;
+        } else {
+            term error = get_default_device_name(driver_data->hostname, global);
+            if (error != OK_ATOM) {
+                free(ssid);
+                free(psk);
+                free(driver_data->hostname);
+                return error;
+            }
+        }
+    } else {
+        int ok = 0;
+        char *buf = malloc(MAX_HOSTNAME_SIZE);
+        if (IS_NULL_PTR(buf)) {
+            free(ssid);
+            free(psk);
+            return OUT_OF_MEMORY_ATOM;
+        } else {
+            buf = interop_term_to_string(hostname_term, &ok);
+        }
+        if (!ok || IS_NULL_PTR(buf)) {
+            free(ssid);
+            free(psk);
+            return BADARG_ATOM;
+        }
+        driver_data->hostname = malloc(strlen(buf) + 1);
```

Comment:
You could use strdup(3) which would do the computation of the length, the allocation and the copy in a single call. Alternatively, if it's not available in some SDKs (it's part of C89), you could save the length to avoid computing it twice.